File: src/main/java/config/Initializer.java
package config;

import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;

public class Initializer extends AbstractAnnotationConfigDispatcherServletInitializer {


    @Override
    protected Class<?>[] getServletConfigClasses() {
        return new Class[] {MvcConfig.class};
    }

    @Override
    protected Class<?>[] getRootConfigClasses() {
        return new Class[] {
                SecurityConfig.class,
                DbConfig.class };
    }


    @Override
    protected String[] getServletMappings() {
        return new String[] {"/api/*"};
    }
}

File: src/main/kotlin/config/Initializer.kt
package config

import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer
import config.MvcConfig

class Initializer : AbstractAnnotationConfigDispatcherServletInitializer() {
    override fun getRootConfigClasses(): Array<Class<*>> {
        return arrayOf()
    }

    override fun getServletConfigClasses(): Array<Class<*>> {
        return arrayOf(MvcConfig::class.java)
    }

    override fun getServletMappings(): Array<String> {
        return arrayOf("/api/*")
    }
}